api/namex/services/cloud_event_messages/cloud_event_messages.py

```diff
@@ -0,0 +1,69 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from datetime import datetime, timedelta, timezone
+import json
+import time
+import uuid
+import nats
+from typing import Optional
+from sentry_sdk import capture_message
+
+from queue_common.messages import create_cloud_event_msg
+from queue_common.service import QueueServiceManager
+from queue_common.service_utils import QueueException, logger
+from config import Config
+
+
+class CloudEventMessageService(object):
+    """Provides services to cloud event message services"""
+    
+    @staticmethod
+    async def sendNameRequestStateEvent(nr_num, old_nr_num, state_cd, old_state_cd):
+        event_loop = asyncio.get_event_loop()
+        qsm = QueueServiceManager()  # pylint: disable=invalid-name
+        await qsm.run(loop=event_loop,
+                      config=Config,
+                      callback=cb_subscription_handler)
+        cloud_event_msg = create_cloud_event_msg(msg_id=str(uuid.uuid4()),
+                                                msg_type='bc.registry.names.request',
+                                                source=f'/requests/{nr_num}',
+                                                time=datetime.
+                                                utcfromtimestamp(time.time()).
+                                                replace(tzinfo=timezone.utc).
+                                                isoformat(),
+                                                identifier=nr_num,
+                                                json_data_body={
+                                                    'request': {
+                                                        'nrNum': nr_num,
+                                                        'oldNrNum': old_nr_num,
```

Comment:
Why is there an old NR num here?
The NRL is and only the NR is there when an invoice is created.
I'm not sure of the flow, but are we going to affiliate NRs that haven't been paid for?

Reply:
I can remove that. I'm just trying to satisfy what Sumesh pointed out: https://github.com/bcgov/namex/pull/1138#discussion_r671594620

Reply:
Please @sumesh-aot confirm we don't need the temporary NRL.

Reply:
@andrepestana-aot @thorwolpert Not sure if NRL is being used now. 
Old and new identifier is needed if the name request uses NRL while creating an invoice, then only we can update the identifier to new identifier to properly show in receipt.

Reply:
An NR doesnt have both an NRL an a NR# so there's no way to map between the 2
It'll depend on if you affiliate an NR that hasn't been paid for yet.
we remove those after 24hrs, so it probably isn't worth it?

Reply:
Affiliation will be based on the NR status change. So I don't think affiliation would happen before payment is received. But the need for old and new was for displaying in receipts and other payment records. 
- If the invoice was created with the NRL identifier, all the pay records would be having NRL. And if we need to update the NRL to NR, the event will need this info (similar to lear where the event contains temp registration number and permanent number).
- Also if we are planning to get the account information from pay records (one of my comment on this PR above) - we won't be able to get it if the records in pay stays NRL.

The second can be solved if the Names stores account-Id and send it in the event.

Reply:
I asked at standup, currently we wont be affiliating NRs that have been paid for, so we don't need to publish events for NRLs

Reply:
@sumesh-aot we're not there yet. it'll take more changes to support both NRL/NR# at the same time, as well as change the pay flow to be more like businesses (create the receipt with the temp reg, then change if paid/completed).

Reply:
Yes, that aligns with my understanding. So do we get account-id in event (if NRL is being used for invoice creation) ?

Reply:
aahh ok. Strike my above comment, it was a race condition :)

Reply:
Looks correct to me.
i don't think it should publish events for states prior to DRAFT (which is the state you enter once paid).